synapse/storage/databases/main/event_federation.py

```diff
@@ -73,12 +73,20 @@
 
 logger = logging.getLogger(__name__)
 
-BACKFILL_EVENT_BACKOFF_UPPER_BOUND_SECONDS: int = int(
-    datetime.timedelta(days=7).total_seconds()
-)
+BACKFILL_EVENT_EXPONENTIAL_BACKOFF_MAXIMUM_DOUBLING_STEPS = 8
 BACKFILL_EVENT_EXPONENTIAL_BACKOFF_STEP_SECONDS: int = int(
     datetime.timedelta(hours=1).total_seconds()
 )
+# The longest backoff period is then:
+# _LONGEST_BACKOFF_PERIOD_SECONDS = (
+#     (1 << BACKFILL_EVENT_EXPONENTIAL_BACKOFF_MAXIMUM_DOUBLING_STEPS)
+#     * BACKFILL_EVENT_EXPONENTIAL_BACKOFF_STEP_SECONDS,
+# )
+# which is 2 ** 8 hours = 256 hours ≈ 10 days. This number needs to fit
+# in a 32 bit signed int, or else Postgres will error.
+# assert _LONGEST_BACKOFF_PERIOD_SECONDS < ((2 ** 31) - 1)
```

Comment:
Seems like we should actually assert this here to avoid the foot-gun in the database

Reply:
Done in a91d8101d